addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf

```diff
@@ -5,188 +5,173 @@
+{   // forEach _allDamages
+    _x params ["_damage", "_bodyPart"];
+
+    // Convert the selectionName to a number and ensure it is a valid selection.
+    private _bodyPartNToAdd = ALL_BODY_PARTS find toLower _bodyPart;
+    if (_bodyPartNToAdd < 0) then {
+        ERROR_1("invalid body part %1",_bodyPart);
+        continue
+    };
 
-            _oldInjury params ["_woundClassIDToAdd", "", "_injuryBleedingRate", "_injuryPain", "", "", "", "_causeLimping", "_causeFracture"];
+    // determine how many wounds to create
+    private _nWounds = [_damage, _thresholds, true] call FUNC(interpolatePoints);
+    if (_nWounds < 1) then {
+        TRACE_2("Damage created zero wounds",_damage,_typeOfDamage);
+        continue
+    };
+    private _dmgPerWound = _damage/_nWounds;
+
+    // find the available injuries for this damage type and damage amount
+    private _weightedWoundTypes = [];
+    {
+        private _weighting = _x select 1;
+        private _woundWeight = [_dmgPerWound, _weighting] call FUNC(interpolatePoints);
+        _weightedWoundTypes pushBack _x;
+        _weightedWoundTypes pushBack _woundWeight;
+    } forEach _damageWoundDetails;
+
+    if (_weightedWoundTypes isEqualTo []) then {
+        TRACE_2("No valid wounds",_damage,_typeOfDamage);
+        continue
+    };
 
-            private _bodyPartNToAdd = if (_isSelectionSpecific) then {_bodyPartN} else {selectRandomWeighted _damageSelectionArray};
+    for "_i" from 1 to _nWounds do {
 
-            _bodyPartDamage set [_bodyPartNToAdd, (_bodyPartDamage select _bodyPartNToAdd) + _woundDamage];
-            _bodyPartVisParams set [[1,2,3,3,4,4] select _bodyPartNToAdd, true]; // Mark the body part index needs updating
+        // Select the injury we are going to add
+        selectRandomWeighted _weightedWoundTypes params ["_woundTypeToAdd", "", "_dmgMultiplier", "_bleedMultiplier", "_sizeMultiplier", "_painMultiplier", "_fractureMultiplier"];
+        GVAR(woundDetails) get _woundTypeToAdd params ["","_injuryBleedingRate","_injuryPain","_causeLimping","_causeFracture"];
+        private _woundClassIDToAdd = GVAR(woundClassNames) find _woundTypeToAdd;
+    
+        // Add a bit of random variance to wounds
+        private _woundDamage = _dmgPerWound * _dmgMultiplier * random [0.9, 1, 1.1];
 
+        _bodyPartDamage set [_bodyPartNToAdd, (_bodyPartDamage select _bodyPartNToAdd) + _woundDamage];
+        _bodyPartVisParams set [[1,2,3,3,4,4] select _bodyPartNToAdd, true]; // Mark the body part index needs updating
 
-            // Damage to limbs/head is scaled higher than torso by engine
-            // Anything above this value is guaranteed worst wound possible
-            private _worstDamage = [2, 1, 4, 4, 4, 4] select _bodyPartNToAdd;
+        // Damage to limbs/head is scaled higher than torso by engine
+        // Anything above this value is guaranteed worst wound possible
+        private _worstDamage = [2, 1, 4, 4, 4, 4] select _bodyPartNToAdd;
```

Comment:
This is weird and I don't like it. First off the comment is wrong - most things tend to do more damage to the torso because:

- It's big and has low penetrability, so projectiles almost never over-penetrate
- It multiplies explosive damage by 6 (presumably to simulate the fact that it's big)

I wonder if whoever tested this did so with armored targets and scaled the values according to that, as it seems to line up with the standard plate carrier, but I don't think it should be balanced like that.

Secondly, because this only affects the wound size and not the underlying damage number it interacts weirdly with thresholds - damage to a limb creates the same number of wounds as if had hit the torso but they are 1/4 the size, so it always creates a lot of small wounds instead of one or two large ones.

Personally I don't think this part scaling is necessary at all, but if it is it should probably be a multiplier to damage before the number of wounds is calculated.

Reply:
@pterolatypus This was added by myself back during the rewrite process, at the time I knew it was a hacky imperfect way to try and resolve the inconsistent wound damage scaling we were seeing (because we didn't understand what was causing it, the armour + over-penetration systems I'm now pretty confident were what we were observing)

The reason it affects wound size was to scale the bleeding rate accordingly since that's the main cause of death from wounding that isn't fatal itself. I have wanted to understand and remove this hacky "fix" ever since first adding it so please be my guest, I think the changes here are a good motivation to do so

Reply:
For now I've replaced this with a flat 2 which I think it reasonable. I've also made a change to the hitpoint macro that affects limb damage.